Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -27,45 +28,82 @@ namespace ROS2
-                auto fileNameToImportAZ(fileNameToImport.toUtf8().constData());
-                m_urdfModel = UrdfParser::ParseFromFile(fileNameToImportAZ);
-                OnModelLoaded();
+                m_robotImporter.Import();
             });
-
-        mainLayout->addStretch();
         setLayout(mainLayout);
     }
 
-    void RobotImporterWidget::OnModelLoaded()
+    void RobotImporterWidget::ReportError(AZStd::string errorMessage)
     {
-        m_robotNameLabel.setText(m_urdfModel->getName().c_str());
+        QMessageBox::critical(this, QObject::tr("Error"), errorMessage.c_str());
+        m_logLabel.setText(errorMessage.c_str());
+        AZ_Error("RobotImporterWidget", false, errorMessage.c_str());
+    }
+    void RobotImporterWidget::ReportWarning(AZStd::string warningMessage)
+    {
+        AZ_Warning("RobotImporterWidget", false, warningMessage.c_str());
+    }
+    void RobotImporterWidget::ReportInfo(AZStd::string infoMessage)
+    {
+        m_logLabel.setText(infoMessage.c_str());
+        AZ_TracePrintf("RobotImporterWidget", infoMessage.c_str());
+    }
+    AZStd::string RobotImporterWidget::GetURDFPath()
+    {
+        return GetPathWithExtension("Unified Robot Description Format (*.urdf)", QFileDialog::ExistingFiles);
+    }
 
-        URDFPrefabMaker prefabMaker(AZStd::string(m_robotFileNameLabel.text().toUtf8().constData()), m_urdfModel);
-        auto outcome = prefabMaker.CreatePrefabFromURDF();
-        if (!outcome)
-        { // TODO - handle, show
-            AZ_Error("RobotImporterWidget", false, "Importing robot definition failed with error: %s", outcome.GetError().c_str());
+    AZStd::string RobotImporterWidget::GetNewPrefabPath()
+    {
+        return GetPathWithExtension("Prefab (*.prefab)", QFileDialog::AnyFile);
+    }
+
+    RobotImporterInputInterface::ExistingPrefabAction RobotImporterWidget::GetExistingPrefabAction()
+    {
+        QMessageBox msgBox;
+        msgBox.setText("The prefab file already exists");
+        msgBox.setInformativeText("Do you want to overwrite it or choose a new filename?");
+        msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel);
+        msgBox.setDefaultButton(QMessageBox::Save);
+        msgBox.setButtonText(QMessageBox::Save, "Overwrite");
+        msgBox.setButtonText(QMessageBox::Discard, "Save As..");
+        auto ret = msgBox.exec();
+        switch (ret)
+        {
+        case QMessageBox::Save:
+            return ExistingPrefabAction::Overwrite;
+        case QMessageBox::Discard:
+            return ExistingPrefabAction::NewName;
+        case QMessageBox::Cancel:
+            return ExistingPrefabAction::Cancel;
+        default:
+            return ExistingPrefabAction::Cancel;
+        }
+    }
+
+    AZStd::string RobotImporterWidget::GetPathWithExtension(AZStd::string extensionDescription, QFileDialog::FileMode mode)
+    {
+        QFileDialog m_importFileDialog;
```

Comment:
misleading name (m_ should be reserved for members)

Reply:
removed `m_` :+1: